test/managers/unit/QueueManager.t.sol

```diff
@@ -0,0 +1,536 @@
+            POOL_A,
+            SC_1,
+            UpdateContractMessageLib.UpdateContractUpdateQueue({minDelay: DEFAULT_MIN_DELAY, extraGasLimit: 0})
+                .serialize()
+        );
+
+        _mockQueuedShares(POOL_A, SC_1, 100, true, 1);
+        _mockQueuedAssets(POOL_A, SC_1, ASSET_1, 100, 0);
+
+        AssetId[] memory assetIds = new AssetId[](1);
+        assetIds[0] = ASSET_1;
+
+        queueManager.sync(POOL_A, SC_1, assetIds);
+
+        vm.warp(block.timestamp + DEFAULT_MIN_DELAY + 1);
+
+        queueManager.sync(POOL_A, SC_1, assetIds);
+    }
+
+    function testSyncWithExtraGasLimit(uint128 extraGasLimit) public {
+        vm.assume(extraGasLimit <= 50_000_000);
+
+        vm.prank(contractUpdater);
+        queueManager.update(
+            POOL_A,
+            SC_1,
+            UpdateContractMessageLib.UpdateContractUpdateQueue({minDelay: 0, extraGasLimit: extraGasLimit}).serialize()
+        );
+
+        _mockQueuedShares(POOL_A, SC_1, 100, true, 1);
+        _mockQueuedAssets(POOL_A, SC_1, ASSET_1, 100, 0);
+
+        AssetId[] memory assetIds = new AssetId[](1);
+        assetIds[0] = ASSET_1;
+
+        vm.expectCall(
+            address(balanceSheet),
+            abi.encodeWithSelector(IBalanceSheet.submitQueuedAssets.selector, POOL_A, SC_1, ASSET_1, extraGasLimit)
+        );
+
+        vm.expectCall(
+            address(balanceSheet),
+            abi.encodeWithSelector(IBalanceSheet.submitQueuedShares.selector, POOL_A, SC_1, extraGasLimit)
+        );
+
+        queueManager.sync(POOL_A, SC_1, assetIds);
+    }
+
+    function testSyncWithDuplicateAssets() public {
+        _mockQueuedShares(POOL_A, SC_1, 100, true, 1);
+        _mockQueuedAssets(POOL_A, SC_1, ASSET_1, 100, 0);
+
+        AssetId[] memory assetIds = new AssetId[](2);
+        assetIds[0] = ASSET_1;
+        assetIds[1] = ASSET_1;
+
+        vm.expectCall(
+            address(balanceSheet),
+            abi.encodeWithSelector(IBalanceSheet.submitQueuedAssets.selector, POOL_A, SC_1, ASSET_1, 0),
+            1
+        );
+
+        queueManager.sync(POOL_A, SC_1, assetIds);
+    }
+
+    function testSyncWithMoreAssetsThanQueued() public {
+        _mockQueuedShares(POOL_A, SC_1, 100, true, 1);
+        _mockQueuedAssets(POOL_A, SC_1, ASSET_1, 100, 0);
+        _mockQueuedAssets(POOL_A, SC_1, ASSET_2, 0, 0);
+        _mockQueuedAssets(POOL_A, SC_1, ASSET_3, 0, 0);
+
+        AssetId[] memory assetIds = new AssetId[](3);
+        assetIds[0] = ASSET_1;
+        assetIds[1] = ASSET_2;
+        assetIds[2] = ASSET_3;
+
+        vm.expectCall(
+            address(balanceSheet),
+            abi.encodeWithSelector(IBalanceSheet.submitQueuedAssets.selector, POOL_A, SC_1, ASSET_1, 0)
+        );
+
+        // No calls for non-queued assets
+        vm.expectCall(
+            address(balanceSheet),
+            abi.encodeWithSelector(IBalanceSheet.submitQueuedAssets.selector, POOL_A, SC_1, ASSET_2, 0),
+            0
+        );
```

Comment:
NIT. Maybe this can be into a `_submitQueuedAssets()` aux method, or even inside `_mockQueuedAssets()`